lib/spack/spack/binary_distribution.py

```diff
@@ -1046,12 +1053,12 @@ def build_tarball(spec, outdir, force=False, rel=False, unsigned=False,
         os.remove('%s.asc' % specfile_path)
 
     web_util.push_to_url(
-        spackfile_path, remote_spackfile_path, keep_original=False)
+        spackfile_path, fetch_spackfile_path, keep_original=False)
```

Comment:
Then here pushing would be done with the push url instead of the fetch one.

Reply:
That's just a mistake. I thought I checked to make sure remote_spackfile_path wasn't used anywhere else. I missed this one.